must-call-checker/src/main/java/org/checkerframework/checker/mustcall/MustCallVisitor.java

```diff
@@ -0,0 +1,152 @@
+   * the set containing top, regardless of what that method returns. This implementation does so.
+   *
+   * @return a set containing only the @MustCallTop annotation
+   */
+  @Override
+  protected Set<? extends AnnotationMirror> getThrowUpperBoundAnnotations() {
+    return Collections.singleton(atypeFactory.TOP);
+  }
+
+  /**
+   * Annotation arguments are treated as return locations for the purposes of defaulting, rather
+   * than parameter locations. This causes them to default incorrectly when the annotation is
+   * defined in bytecode. See https://github.com/typetools/checker-framework/issues/3178 for an
+   * explanation of why this is necessary to avoid false positives.
+   *
+   * <p>Skipping this check in the MustCall checker is safe, because the MustCall checker is not
+   * concerned with annotation arguments (which must be literals, and therefore won't have (or be
+   * able to fulfill) must-call obligations).
+   */
+  @Override
+  public Void visitAnnotation(AnnotationTree node, Void p) {
+    return null;
+  }
+
+  /**
+   * Only check type arguments that are bottom, to avoid false positives.
+   *
+   * <p>Without this code, the checker issues false positives at any point where a type with a
+   * non-default @MustCall annotation is used as a type argument. For example, if the user writes
+   * {@code List<Socket>} (and Socket is must-call close), then without this code a
+   * type.argument.type.incompatible error is issued. These errors are spurious, because the List of
+   * Sockets does in fact have a responsibility to close each of those sockets.
+   */
+  protected void checkTypeArguments(
+      Tree toptree,
+      List<? extends AnnotatedTypeParameterBounds> paramBounds,
+      List<? extends AnnotatedTypeMirror> typeargs,
+      List<? extends Tree> typeargTrees,
+      Name typeOrMethodName,
+      List<?> paramNames) {
+    List<Integer> indicesToRemove = new ArrayList<>();
+    for (int i = 0; i < typeargs.size(); i++) {
+      AnnotatedTypeMirror atm = typeargs.get(i);
+      if (containsNonDefaultAnnotation(atm)) {
+        indicesToRemove.add(i);
+      }
+    }
+    // The lists are unmodifiable, so new lists have to be constructed
+    // rather than removing from the old ones.
+    List<AnnotatedTypeParameterBounds> paramBoundsWithoutNonDefault = new ArrayList<>();
+    List<AnnotatedTypeMirror> typeargsWithoutNonDefault = new ArrayList<>();
+    for (int i = 0; i < typeargs.size(); i++) {
+      if (!indicesToRemove.contains(i)) {
+        paramBoundsWithoutNonDefault.add(paramBounds.get(i));
+        typeargsWithoutNonDefault.add(typeargs.get(i));
+      }
+    }
+    super.checkTypeArguments(
+        toptree,
+        paramBoundsWithoutNonDefault,
+        typeargsWithoutNonDefault,
+        typeargTrees,
+        typeOrMethodName,
+        paramNames);
+  }
+
+  /**
+   * Does a deep scan of the annotated type for any non-default (that is, non-bottom) annotation.
```

Comment:
Non-default and non-bottom are different.  The default is `@CalledMethods({})`.  Bottom is `@CalledMethods({a, b, c, d, ...})`.